### PR TITLE
3014 - Outdated submissions banner

### LINKED ISSUE #3014: Outdated info banner for submission history results with data processed before 5/31/24
**Description:**

This ticket delivers an info banner appearing contextually at the top of any submission history page containing error reports generated prior to May 31 2024.

Original description:
```The ATO for the legacy system (TDRS) is set to expire at the end of May 2024. This means that files will no longer be processed thru TDRS and TDRS feedback reports will not be sent (as of 5/31/2024). 

To-date, STTs have not been relying on the TDP-generated feedback reports, primarily because these feedback reports have changed overtime as the parsing and validation engine has been refined. To help mitigate confusion, OFA recommends wiping the feedback-related columns in the submission history for files submitted prior to 5/31/2024.  
```

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [x] Banner appears on submission history if 1 or more files are present for viewed quarter that were submitted prior to May 31, 2024
- [x] Banner does not appear if a given quarter doesn't have data submitted before May 31 2024 
- [x] Language matches spec "```Please note that error reports and submission history content for files submitted prior to
May 31, 2024 may be outdated. Please resubmit to get access to updated information.```"
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 

- [ ] Create logic to check for submission history entries made before 5/31/24
- [ ] Implement info banner when that logic returns true
- [ ] Tag for a11y review
- [ ] Run Testing Checklist and confirm all tests pass

**Notes:**
- This ticket is particularly aimed at smoothing the transition for Tribes who aren't used to having the level of data feedback submission history provides. Requesting that STTs simply ignore inaccurate older data has not been particularly successful at alleviating concerns.

**Supporting Documentation:**
[Figma](https://www.figma.com/design/irgQPLTrajxCXNiYBTEnMV/TDP-Mockups-For-Feedback?node-id=8993-259&t=OQgW2zgCgvJaKsFL-1)

image

### PR DESCRIPTION
**Pull request by jtimpe**
## Summary of Changes
Pull request closes #3014 
* ~~adds outdated submissions banner for files submitted before 5/31/2024~~
* ~~removes the error report link for those files (replaces with "Not Available")~~
+ combine "submitted by" and "submitted on"

## How to Test
```
cd tdrs-backend && docker compose up
cd tdrs-frontend && docker compose up --build
```

1. Open http://localhost:3000/ and sign in.
1. Submit a data file. Make sure the banner does not appear, and the error reports download link is shown.
1. Modify the `created_at` field of the datafile to before 5/31/2024
   ```bash
   docker compose exec web python manage.py shell_plus

   f = DataFile.objects.all().first()
   from django.utils import timezone
   from datetime import datetime
   f.created_at = datetime(2023, 10, 11)
   f.save()
   ```
1. Return to the datafiles page. Ensure the banner displays and the error report download link is replaced with "Not Available"

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
~~+ [ ] Banner appears on submission history if 1 or more files are present for viewed quarter that were submitted prior to May 31, 2024~~
~~+ [ ] Banner does not appear if a given quarter doesn't have data submitted before May 31 2024~~
~~+ [ ] Language matches spec "Please note that error reports and submission history content for files submitted prior to May 31, 2024 may be outdated. Please resubmit to get access to updated information."~~
+ [x] combine "submitted by" and "submitted on"
+ [ ] Testing Checklist has been run and all tests pass
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [x] Does this PR complete the epic? 
+ [x] Are links included to any other gov-approved PRs associated with epic?
+ [x] Does PR include documentation for Raft's a11y review? 
+ [x] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

**Review by raftmsohani — APPROVED** [review]

**Review by andrew-jameson — APPROVED** [review]

File: tdrs-frontend/src/components/SubmissionHistory/SubmissionHistory.jsx [hunk 1]
```diff
@@ -71,8 +81,33 @@ const SubmissionHistory = ({ filterValues }) => {
...
   }, [hasFetchedFiles, files, dispatch, filterValues])
 
+  let submissionThreshold = new Date('05-31-2024')
+  const fileIsOutdated = (f) => {
+    let created_date = new Date(f.createdAt)
+    return created_date < submissionThreshold
+  }
+
+  const hasOutdatedSubmissions = () =>
+    files.some((e, i, a) => fileIsOutdated(e))
```
Comment (elipe17) [inline tdrs-frontend/src/components/SubmissionHistory/SubmissionHistory.jsx]: Could we give the params of the functor a bit more descriptive names?

**Review by elipe17 — APPROVED** [review]
One comment, doesn't halt anything. LGTM

File: tdrs-frontend/src/components/SubmissionHistory/SubmissionHistory.jsx (hunk 1 above)
Comment (jtimpe) [reply tdrs-frontend/src/components/SubmissionHistory/SubmissionHistory.jsx]: javascript bad habits 😄

**Comment by victoriaatraft** [comment]
LGTM @jtimpe 
Screenshot 2024-10-15 at 11 24 48 AM

**Review by victoriaatraft — APPROVED** [review]

**Comment by ADPennington** [comment]
> LGTM @jtimpe Screenshot 2024-10-15 at 11 24 48 AM

per standup today, @jtimpe noted that submitted by column is blank because his user's first and last name is blank in user profile.

**Review by ADPennington — COMMENTED** [review]
@jtimpe this PR delivers the following expected results 🥇 :

- Banner appears when files were submitted prior to May 31, 2024 ✔️ 
- error report is unavailable for abovementioned files ✔️ 

However, we believe this change is now out-of-scope given the reparsing work. In particular:

- when files are reparsed, the submission history metadata, including the error reports, could be refreshed, but if the reparsed file was submitted prior to May 31, 2024, this report will still be unavailable (see evidence below)

reparse3014

- The data team considered some possible conditions to apply that could help make this workable, including:
  - changing the date to something much earlier (e.g. Dec 31, 2023)
  - adding a condition to display the error report if the file has been reparsed
  - making the error report unavailable if a file has been resubmitted

All of these alternatives scope-creep the ticket and adds unnecessary complexity to an otherwise great solution to a problem (i.e. STTs reviewing and asking Qs about outdated reports) that we think is now resolved with the reparsing feature. 

cc: @ttran-hub @klinkoberstar @lfrohlich @vlasse86

**Comment by jtimpe** [comment]
@ADPennington we could potentially make use of the new `ReparseFileMeta` to see if a particular file has newer reparses available. I have to think more about how that would work, and it definitely increases scope a bit, but gut feeling says it should be doable.

**Comment by jtimpe** [comment]
@ADPennington i made the change to include information about the datafile's reparses inside of the api response. The Submission History page now checks the `ReparseFileMeta`'s `finished_at` field if the original file submission date is outdated. If there is a recent reparse, it will not show the banner and the error report will be available. If there is no recent reparse, it will show the banner and disable the report. 

One caveat is that the Submission History page maintains the original created date. So, if that date is prior to May 31 2024, but the file has a more recent reparse, it may be slightly confusing. Ideally there would be a way to show the file's version history a little more clearly with respect to reparse. Open to suggestions on that front.

**Comment by ADPennington** [comment]
> @ADPennington i made the change to include information about the datafile's reparses inside of the api response. The Submission History page now checks the `ReparseFileMeta`'s `finished_at` field if the original file submission date is outdated. If there is a recent reparse, it will not show the banner and the error report will be available. If there is no recent reparse, it will show the banner and disable the report.
> 
> One caveat is that the Submission History page maintains the original created date. So, if that date is prior to May 31 2024, but the file has a more recent reparse, it may be slightly confusing. Ideally there would be a way to show the file's version history a little more clearly with respect to reparse. Open to suggestions on that front.

thank you for this update @jtimpe -- I plan to retest with @ttran-hub and @klinkoberstar early next week. in the meantime, if @victoriaatraft and @reitermb would like to explore this solution in a deployed environment, please feel free 😄

File: tdrs-backend/tdpservice/data_files/serializers.py [hunk 2]
```diff
@@ -56,6 +88,10 @@ def get_has_error(self, obj):
         parser_errors = ParserError.objects.filter(file=obj.id)
         return len(parser_errors) > 0
 
+    def get_reparse_file_metas(self, instance):
```
Comment (jtimpe) [inline tdrs-backend/tdpservice/data_files/serializers.py]: convert to bool - "has recent reparse"

**Comment by jtimpe** [comment]
> So, if that date is prior to May 31 2024, but the file has a more recent reparse, it may be slightly confusing. Ideally there would be a way to show the file's version history a little more clearly with respect to reparse.

could potentially just show the last reparse date, or another column stating when the file was last reparsed

**Comment by ADPennington** [comment]
> > So, if that date is prior to May 31 2024, but the file has a more recent reparse, it may be slightly confusing. Ideally there would be a way to show the file's version history a little more clearly with respect to reparse.
> 
> 
> 
> could potentially just show the last reparse date, or another column stating when the file was last reparsed

I like the idea of adding another column to the submission history table for "Most Recent Processing Date". I assume it would be the submission date for files never re-parsed. And the reparse date otherwise? thoughts @jtimpe @klinkoberstar @ttran-hub? also interested in a11y implications @victoriaatraft

**Comment by klinkoberstar** [comment]
> > > So, if that date is prior to May 31 2024, but the file has a more recent reparse, it may be slightly confusing. Ideally there would be a way to show the file's version history a little more clearly with respect to reparse.
> > 
> > 
> > could potentially just show the last reparse date, or another column stating when the file was last reparsed
> 
> I like the idea of adding another column to the submission history table for "Most Recent Processing Date". I assume it would be the submission date for files never re-parsed. And the reparse date otherwise? thoughts @jtimpe @klinkoberstar @ttran-hub? also interested in a11y implications @victoriaatraft

I really like the idea of providing STTs some transparency into when their files were last processed with this column!

**Comment by jtimpe** [comment]
@ADPennington @klinkoberstar @victoriaatraft @reitermb 

came up with a couple different options.

1. extra column - if we go this route, what should the default/empty state be?
   image
2. add another line to the 'submitted on' column - my thought was this could evolve into a dropdown list of all past reparse events (but it just shows the latest right now)
   image

open to any other ideas as well! i pushed a combo of both changes so you guys can deploy and compare

**Comment by ADPennington** [comment]
> @ADPennington @klinkoberstar @victoriaatraft @reitermb
> 
> came up with a couple different options.
> 
> 1. extra column - if we go this route, what should the default/empty state be?
>    image
> 2. add another line to the 'submitted on' column - my thought was this could evolve into a dropdown list of all past reparse events (but it just shows the latest right now)
>    image
> 
> open to any other ideas as well! i pushed a combo of both changes so you guys can deploy and compare
@jtimpe @victoriaatraft 
- we like option 1
- column label suggestion: "Most Recent Processing Date"
- default value for most recent processing date == submission date
- @klinkoberstar and I want to think more about the banner language. We want to ensure that we arent recommending STTs to resubmit old versions of data.

**Comment by reitermb** [comment]
> > @ADPennington @klinkoberstar @victoriaatraft @reitermb
> > came up with a couple different options.
> > 
> > 1. extra column - if we go this route, what should the default/empty state be?
> >    image
> > 2. add another line to the 'submitted on' column - my thought was this could evolve into a dropdown list of all past reparse events (but it just shows the latest right now)
> >    image
> > 
> > open to any other ideas as well! i pushed a combo of both changes so you guys can deploy and compare
> > @jtimpe @victoriaatraft
> 
> * we like option 1
> * column label suggestion: "Most Recent Processing Date"
> * default value for most recent processing date == submission date
> * @klinkoberstar and I want to think more about the banner language. We want to ensure that we arent recommending STTs to resubmit old versions of data.

For the near term we also lean option 1, but @victoriaatraft and I are also curious whether there's potential to do away both with the need for a banner and for the extra column. i.e. when reparsing occurs could the system treat it more like a resubmission? Create a new line item, maybe denote in Submitted by that it was a system action or an OFA action. Eliminate the need to shift aggregate data, statuses, or the contents of the error report.

**Comment by ADPennington** [comment]
> > > @ADPennington @klinkoberstar @victoriaatraft @reitermb
> 
> > > came up with a couple different options.
> 
> > > 
> 
> > > 1. extra column - if we go this route, what should the default/empty state be?
> 
> > >    image
> 
> > > 2. add another line to the 'submitted on' column - my thought was this could evolve into a dropdown list of all past reparse events (but it just shows the latest right now)
> 
> > >    image
> 
> > > 
> 
> > > open to any other ideas as well! i pushed a combo of both changes so you guys can deploy and compare
> 
> > > @jtimpe @victoriaatraft
> 
> > 
> 
> > * we like option 1
> 
> > * column label suggestion: "Most Recent Processing Date"
> 
> > * default value for most recent processing date == submission date
> 
> > * @klinkoberstar and I want to think more about the banner language. We want to ensure that we arent recommending STTs to resubmit old versions of data.
> 
> 
> 
> For the near term we also lean option 1, but @victoriaatraft and I are also curious whether there's potential to do away both with the need for a banner and for the extra column. i.e. when reparsing occurs could the system treat it more like a resubmission? Create a new line item, maybe denote in Submitted by that it was a system action or an OFA action. Eliminate the need to shift aggregate data, statuses, or the contents of the error report. 

It does seem like we're at a juncture where the value of the banner is in question and need to decide a path forward. I do like where we're going in terms of adding evidence of reparsing to submission history though. I recommend we have a sync to make final decisions on this PR. @jtimpe has provided some great options to consider.

**Comment by reitermb** [comment]
> > > > @ADPennington @klinkoberstar @victoriaatraft @reitermb
> > 
> > > > came up with a couple different options.
> > 
> > > > 
> > 
> > > > 1. extra column - if we go this route, what should the default/empty state be?
> > 
> > > >    image
> > 
> > > > 2. add another line to the 'submitted on' column - my thought was this could evolve into a dropdown list of all past reparse events (but it just shows the latest right now)
> > 
> > > >    image
> > 
> > > > 
> > 
> > > > open to any other ideas as well! i pushed a combo of both changes so you guys can deploy and compare
> > 
> > > > @jtimpe @victoriaatraft
> > 
> > > 
> > 
> > > * we like option 1
> > 
> > > * column label suggestion: "Most Recent Processing Date"
> > 
> > > * default value for most recent processing date == submission date
> > 
> > > * @klinkoberstar and I want to think more about the banner language. We want to ensure that we arent recommending STTs to resubmit old versions of data.
> > 
> > 
> > 
> > For the near term we also lean option 1, but @victoriaatraft and I are also curious whether there's potential to do away both with the need for a banner and for the extra column. i.e. when reparsing occurs could the system treat it more like a resubmission? Create a new line item, maybe denote in Submitted by that it was a system action or an OFA action. Eliminate the need to shift aggregate data, statuses, or the contents of the error report. 
> 
> It does seem like we're at a juncture where the value of the banner is in question and need to decide a path forward. I do like where we're going in terms of adding evidence of reparsing to submission history though. I recommend we have a sync to make final decisions on this PR. @jtimpe has provided some great options to consider. 
> 
> 

Would spinning something up tomorrow sound good? Or maybe a post stand-up conversation Friday

**Comment by victoriaatraft** [comment]
> > > > > @ADPennington @klinkoberstar @victoriaatraft @reitermb
> > > 
> > > 
> > > > > came up with a couple different options.
> > > 
> > > 
> > > > >
> > > 
> > > 
> > > > > 1. extra column - if we go this route, what should the default/empty state be?
> > > 
> > > 
> > > > > image
> > > 
> > > 
> > > > > 2. add another line to the 'submitted on' column - my thought was this could evolve into a dropdown list of all past reparse events (but it just shows the latest right now)
> > > 
> > > 
> > > > > image
> > > 
> > > 
> > > > >
> > > 
> > > 
> > > > > open to any other ideas as well! i pushed a combo of both changes so you guys can deploy and compare
> > > 
> > > 
> > > > > @jtimpe @victoriaatraft
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > * we like option 1
> > > 
> > > 
> > > > * column label suggestion: "Most Recent Processing Date"
> > > 
> > > 
> > > > * default value for most recent processing date == submission date
> > > 
> > > 
> > > > * @klinkoberstar and I want to think more about the banner language. We want to ensure that we arent recommending STTs to resubmit old versions of data.
> > > 
> > > 
> > > For the near term we also lean option 1, but @victoriaatraft and I are also curious whether there's potential to do away both with the need for a banner and for the extra column. i.e. when reparsing occurs could the system treat it more like a resubmission? Create a new line item, maybe denote in Submitted by that it was a system action or an OFA action. Eliminate the need to shift aggregate data, statuses, or the contents of the error report.
> > 
> > 
> > It does seem like we're at a juncture where the value of the banner is in question and need to decide a path forward. I do like where we're going in terms of adding evidence of reparsing to submission history though. I recommend we have a sync to make final decisions on this PR. @jtimpe has provided some great options to consider.
> 
> Would spinning something up tomorrow sound good? Or maybe a post stand-up conversation Friday

I'm good either way. My calendar is pretty open tomorrow @jtimpe @klinkoberstar @ADPennington @reitermb

**Comment by ADPennington** [comment]
@reitermb @victoriaatraft @jtimpe @klinkoberstar 11/1 after standup sounds good. Does this work for you all?

**Comment by klinkoberstar** [comment]
> @reitermb @victoriaatraft @jtimpe @klinkoberstar 11/1 after standup sounds good. Does this work for you all?

I have a conflict but I should be able to move it - tentative yes from me!

**Comment by victoriaatraft** [comment]
> > @reitermb @victoriaatraft @jtimpe @klinkoberstar 11/1 after standup sounds good. Does this work for you all?
> 
> I have a conflict but I should be able to move it - tentative yes from me!

Sending invite!

**Comment by ADPennington** [comment]
@reitermb will add notes here on recommendations to rescope this PR and follow-on tickets cc: @jtimpe @victoriaatraft @klinkoberstar @vlasse86

**Comment by reitermb** [comment]
Action-relevant notes from Nov 1 sync:

- Jan & UX to schedule a tag up to work an iteration of Jan's proposed Version B into this PR. 
- @ADPennington and @klinkoberstar, do you have thoughts on whether we only need to surface the most recent reprocessing date to users or whether there's value in the ability to surface a complete log of reprocessing events for a given file?
- UX to spin up an email templating ticket to create call to action language for users around reprocessing events
- UX to spin up a ticket to spec out / ideate on the idea of a "this file needs reprocessing" feature for STT users to allow them to attract OFA attention to specific data files which may need attention. 

cc @ADPennington @klinkoberstar @jtimpe @victoriaatraft

**Comment by ADPennington** [comment]
> Action-relevant notes from Nov 1 sync:
> 
> * Jan & UX to schedule a tag up to work an iteration of Jan's proposed Version B into this PR.
> * @ADPennington and @klinkoberstar, do you have thoughts on whether we only need to surface the most recent reprocessing date to users or whether there's value in the ability to surface a complete log of reprocessing events for a given file?

hmm; good q! I could be convinced either way, but I'm currently leaning toward only the most recent re-processing event. I don't expect this to happen often per file, and I'm not sure why we would need to surface a full history of what feels a bit like system updates? If there's a compelling reason, do we think a count of re-processing events would be helpful? 

> * UX to spin up an email templating ticket to create call to action language for users around reprocessing events
> * UX to spin up a ticket to spec out / ideate on the idea of a "this file needs reprocessing" feature for STT users to allow them to attract OFA attention to specific data files which may need attention.
> 
> cc @ADPennington @klinkoberstar @jtimpe @victoriaatraft

**Comment by victoriaatraft** [comment]
Recap November 5 Internal Sync @jtimpe @reitermb 

- Move forward with Option B
- Add Tooltip for report not available to help call out information presented in the banner.
- @jtimpe Have the text in the column also be included in the hover target. 
- Text: `This file was submitted prior to May 31, 2024. Please resubmit to get access to updated information`
- UX will do a quick screen for screenreader

 Created #3271 as separate enhancement ticket because it's outside scope:
- Explore combining Submitted On and Submitted By columns. To follow up: show indication that it is a System Action for Reprocessed On
- Create mockup to remove Reprocessed On column because that is inherent to option B. Simplify the data that is Submitted on and that triggers the dialogue box or modal to get the dates and history of reprocessing.

cc: @ADPennington @klinkoberstar @lfrohlich

**Comment by ADPennington** [comment]
> Recap November 5 Internal Sync @jtimpe @reitermb
> 
> * Move forward with Option B

- can you show us a screenshot of Option B ? is it the one referenced above with "reprocessed date" in the same column as original submission date? If so, could you remind us of the benefits of not having a separate column for this? 

- we really like the idea of having a "most recently processed date" somewhere in the submission history because it shows evidence of things we've promised the STTs that we would do (which is to re-process files). 


> * Add Tooltip for report not available to help call out information presented in the banner.

- could you remind us what the goal of the tooltip is? We thought it would be used to describe the new info in submission history (i.e. most recently processed date). 

> * @jtimpe Have the text in the column also be included in the hover target.
> * Text: `This file was submitted prior to May 31, 2024. Please resubmit to get access to updated information`
> * UX will do a quick screen for screenreader

- did we agree to keep this banner? If so, why? We're concerned that this would encourage STTs to overwrite more recently submitted data. 
- With reparsing, we do not yet see evidence that any STTs need to resubmit to get an updated error report. 

> Created #3271 as separate enhancement ticket because it's outside scope:
> * Explore combining Submitted On and Submitted By columns. To follow up: show indication that it is a System Action for Reprocessed On
> * Create mockup to remove Reprocessed On column because that is inherent to option B. Simplify the data that is Submitted on and that triggers the dialogue box or modal to get the dates and history of reprocessing.

- Ideally, we could pick up #3271 after there is evidence from STTs or sys admins that a full history of re-processing is valuable for STTs. Currently, too early to say. 

> cc: @ADPennington @klinkoberstar @lfrohlich

cc: @klinkoberstar @lfrohlich @jtimpe @reitermb @victoriaatraft

**Comment by victoriaatraft** [comment]
@ADPennington  
cc: @klinkoberstar @lfrohlich @jtimpe @reitermb  

Would it be helpful if we stay on a few minutes after standup tomorrow to go through the Q's?

**Comment by klinkoberstar** [comment]
> @ADPennington cc: @klinkoberstar @lfrohlich @jtimpe @reitermb
> 
> Would it be helpful if we stay on a few minutes after standup tomorrow to go through the Q's?

I have a conflict at that time but can catch up with Alex after if that's the best time!

**Review by victoriaatraft — APPROVED** [review]
LGTM

**Comment by ADPennington** [comment]
will test the following:

- no banner
- submitted on and submitted by merged into one column

anything else? @jtimpe (asking because i think the summary may now be outdated 😄 )

**Comment by jtimpe** [comment]
@ADPennington 
>     * no banner
>     * submitted on and submitted by merged into one column
> anything else? @jtimpe (asking because i think the summary may now be outdated 😄 )

* the error report download link will also display regardless of if the file is outdated.
* i also snuck in `file.latest_reparse_meta` to the api. so any changes we want to make to the frontend in the future (showing latest reprocessed date, for example) can be an easier change. it's not used anywhere currently, though

updated the description, thank you for calling that out!

File: tdrs-backend/tdpservice/data_files/serializers.py [hunk 3]
```diff
@@ -12,8 +12,26 @@
...
 logger = logging.getLogger(__name__)
 
+
+class ReparseFileMetaSerializer(serializers.ModelSerializer):
+    """Serializer for ReparseFileMeta class."""
+
+    class Meta:
+        """Meta class."""
+
+        model = ReparseFileMeta
```
Comment (ADPennington) [inline tdrs-backend/tdpservice/data_files/serializers.py:26]: ✔️ 

Screenshot 2024-11-15 092322

File: tdrs-frontend/src/components/SubmissionHistory/CaseAggregatesTable.jsx [hunk 4]
```diff
@@ -24,17 +26,13 @@ const MonthSubRow = ({ data }) =>
...
   return (
     <>
       <tr>
         <th scope="rowgroup" rowSpan={3}>
-          {formatDate(file.createdAt)}
-        </th>
-
-        <th scope="rowgroup" rowSpan={3}>
-          {file.submittedBy}
+          {formatDate(file.createdAt) + ' by ' + file.submittedBy}
```
Comment (ADPennington) [inline tdrs-frontend/src/components/SubmissionHistory/CaseAggregatesTable.jsx:34]: ✔️ 

Screenshot 2024-11-15 092600

File: tdrs-frontend/src/components/SubmissionHistory/TotalAggregatesTable.jsx [hunk 5]
```diff
@@ -22,17 +24,13 @@ const MonthSubRow = ({ data }) =>
...
   return (
     <>
       <tr>
         <th scope="rowgroup" rowSpan={3}>
-          {formatDate(file.createdAt)}
-        </th>
-
-        <th scope="rowgroup" rowSpan={3}>
-          {file.submittedBy}
+          {formatDate(file.createdAt) + ' by ' + file.submittedBy}
```
Comment (ADPennington) [inline tdrs-frontend/src/components/SubmissionHistory/TotalAggregatesTable.jsx:32]: Screenshot 2024-11-15 091822_LI

**Review by ADPennington — APPROVED** [review]
> > * no banner

confirmed ✔️ 
Screenshot 2024-11-15 093441

> > * submitted on and submitted by merged into one column
confirmed ✔️ 

> * the error report download link will also display regardless of if the file is outdated.

confirmed ✔️ 

> * i also snuck in `file.latest_reparse_meta` to the api. so any changes we want to make to the frontend in the future (showing latest reprocessed date, for example) can be an easier change. it's not used anywhere currently, though

confirmed ✔️

**Pull request merged**